src/parser/constant_construct.rs

```diff
@@ -19,9 +23,48 @@ impl ConstantConstruct {
     }
 
     pub(crate) fn string_constant(input: &str) -> ParseResult<&str, Box<dyn Instruction>> {
-        let (input, string_value) = Token::string_constant(input)?;
+        let (input, inner) = preceded(Token::double_quote, ConstantConstruct::inner_string)(input)?;
+        let string = inner.unwrap_or_else(|| Box::new(JkString::from("")));
+
+        Ok((input, string))
+    }
+
+    /// inner_string = '"'
+    ///              | '{' expr '}' string
+    ///              | CHAR string
+    fn inner_string(input: &str) -> ParseResult<&str, Option<Box<dyn Instruction>>> {
+        let special: &[_] = &['"', '{'];
+
+        if let Ok((input, _)) = Token::double_quote(input) {
+            Ok((input, None))
```

Comment:
So if the first character we see is a double quote, we return `None`? Is that why we unwrap_or_else in the caller?

Reply:
The elegant implementation here was to return an empty jkstring when we reach a double quote, that way we indiscriminately concat whatever we have afterwards instead of matching as you can see in the concat function, it would also avoid the unwrap_or_else in the function above.
However doing this cause EVERY string to call .concat("") when built, it also causes extra heap allocation in the AST to create this method call.
That's why instead i return an Option and handle it as a special case, thus avoiding unnecessary heap allocation and slightly better runtime performances.